tensorflow/go/op/gradients.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package op
+
+import tf "github.com/tensorflow/tensorflow/tensorflow/go"
+
+// Gradients adds gradients computation ops to the graph according to scope.
+//
+// Arguments:
+//  prefix: unique string prefix applied before the names of nodes added to the graph to
+//    compute gradients. If null, will use "Gradients".
+//  y: output of the function to derive
+//  x: inputs of the function for which partial derivatives are computed
+//  dx: if not null, the partial derivatives of some loss function L w.r.t. y
+//
+//  return the partial derivatives
+func Gradients(scope *Scope, prefix string, y []tf.Output, x []tf.Output, dx ...tf.Output) (output []tf.Output) {
```

Comment:
Can we avoid a `prefix` argument here altogether, that will be more consistent with other ops (where the name for the node in the graph isn't an argument)

Reply:
I'm not sure.
The meaning of `namespace` in `Scope` is not completely same with `prefix` for gradient.

For example:
If the `namespace` is `sub`, the  `Op.name()` of the `gradients` should be `sub/Gradients`;
But if we need a `sub/my_gradients`, there isn't other methods except for a `prefix` argument...

The JAVA implementation adds a `opName` to `Scope` structure(to be exact, it's a new `NameScope` structure). So they can add customized prefix by change change the value of `opName`.

If we actually don't want a `prefix` argument, we have two solutions (I see):
1. Always use `Gradients` as prefix, such as `Gradients`, `sub/Gradients`, `sub/Gradients_1`, etc

2. Add `opName` to `Scope` structure(also related methods like `SubScopeWithName`(to new a scope with customized opName) and `opName`(need to change behavior). It will be same with java implementation.